front/components/agent_builder/AgentBuilderLayout.tsx

```diff
@@ -99,17 +100,31 @@ export function AgentBuilderLayout({
                   collapsible={true}
                   onCollapse={handlePanelCollapse}
                   onExpand={handlePanelExpand}
-                  className={
+                  className={cn(
                     !isResizing
                       ? "overflow-hidden transition-all duration-300 ease-in-out"
-                      : "overflow-hidden"
-                  }
+                      : "overflow-hidden",
+                    !isPreviewPanelOpen && "hidden md:block",
+                    isPreviewPanelOpen && "md:relative",
+                    isPreviewPanelOpen &&
+                      "absolute inset-0 z-50 bg-background md:relative md:inset-auto md:z-auto"
```

Comment:
Same let's ditch the z index here if possible.

Also can we use a tertiary instead condition?